avm/res/devops-infrastructure/managed-devops-pools/main.bicep

```diff
@@ -0,0 +1,448 @@
+@allowed([
+  'Interactive'
+  'Service'
+])
+param logonType string?
+
+@description('Required. Defines how the machine will be handled once it executed a job.')
+param agentProfile agentProfileType
+
+@description('Required. Defines the organization in which the pool will be used.')
+param organizationProfile organizationProfileType
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. The diagnostic settings of the service.')
+param diagnosticSettings diagnosticSettingType
+
+@description('Optional. The managed service identities assigned to this resource.')
+@metadata({
+  example: '''
+  {
+    systemAssigned: true,
+    userAssignedResourceIds: [
+      '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myManagedIdentity'
+    ]
+  }
+  {
+    systemAssigned: true
+  }
+  '''
+})
+param managedIdentities managedIdentitiesType
+
+var builtInRoleNames = {
+  Contributor: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+  Owner: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
+  Reader: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+  'Role Based Access Control Administrator (Preview)': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'f58310d9-a9f6-439a-9e8d-f62e7b41a168'
+  )
+  'User Access Administrator': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '18d7d88d-d35e-4fb5-a5c3-7773c20a72d9'
+  )
+}
+
+var formattedUserAssignedIdentities = reduce(
+  map((managedIdentities.?userAssignedResourceIds ?? []), (id) => { '${id}': {} }),
+  {},
+  (cur, next) => union(cur, next)
+) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
+
+var identity = !empty(managedIdentities)
+  ? {
+      type: (managedIdentities.?systemAssigned ?? false)
+        ? (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned')
+        : (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'UserAssigned' : 'None')
+      userAssignedIdentities: !empty(formattedUserAssignedIdentities) ? formattedUserAssignedIdentities : null
+    }
+  : null
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.di-mdp.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
```

Comment:
This telemetry id looks quite ... off 😄

Reply:
What is the way to go here? I copy/pasted it from another module and abbreviated the devops-infrastructure to di and managed-devops-pools to mdp else the name would be too long (64 max. length when not abbreviated is 69).

Reply:
There's a process behind it. @matebarabas or another colleague would need to add the module to an internal store and would then provide you with the correct ID via the module proposal if I'm not mistaken

Reply:
That's correct. The module indexes just have been updated; please check them for all mandatory values (GH team names, etc.). For your convenience, the value you should use for the telemetry ID, is this one: `46d3xbcp.res.devopsinfrastructure-pool`.